### PR TITLE
Docfx

### PR DESCRIPTION
**Pull request by pepone**
This PR adds a BuildDoc target that generates HTML docs from C# code using docfx

Install docfx from choco/brew (https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html)

To build the docs try
```
dotnet msbuild msbuild/ice.proj /t:BuildDoc
docfx serve _site
```

### CODE REVIEW

File: csharp/index.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# This is the **HOMEPAGE**.
+Refer to [Markdown](http://daringfireball.net/projects/markdown/) for how to write markdown files.
+## Quick Start Notes:
+1. Add images to the *images* folder if the file is referencing an image.
```
Comment (externl) [inline csharp/index.md:4]: I don't see this file being included anywhere in `docfx.json`

File: .gitignore [hunk 2]
```diff
@@ -123,3 +123,6 @@ Carthage
 
 # Log files generated by the test suite.
 *.log
+
+# docfx
+csharp/_site
```
[the remark this answers is not in the document]
Comment (externl) [reply .gitignore:128]: Actually,  this should be moved into `csharp/.gitignore`

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Review by externl — APPROVED** [review]

**Comment by pepone** [comment]
I thinking to do some updates to move the docfx files to csharp/doc folder

**Comment by externl** [comment]
> I thinking to do some updates to move the docfx files to csharp/doc folder

Good idea!

File: .gitignore [hunk 3]
```diff
@@ -125,4 +125,4 @@ Carthage
 *.log
 
 # docfx
-csharp/_site
\ No newline at end of file
+csharp/doc/_site
```
Comment (externl) [inline .gitignore:128]: I think we should move this to `csharp/.gitignore`

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (pepone) [reply .gitignore]: fixed

File: cpp/src/slice2cs/Gen.cpp [hunk 4]
```diff
@@ -3075,6 +3075,8 @@ Slice::Gen::ClassFactoryVisitor::visitClassDefStart(const ClassDefPtr& p)
     string name = fixId(p->name());
     _out << sp;
     emitCommonAttributes();
+    _out << nl << "[global::System.ComponentModel.EditorBrowsable("
+         << "global::System.ComponentModel.EditorBrowsableState.Never)]";
```
[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp]: Yes I added `emitEditorBrowsableNeverAttribute`

File: csharp/api/.gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+###############
+#  temp file  #
+###############
```
[the remark this answers is not in the document]
Comment (pepone) [reply csharp/api/.gitignore]: Fixed

File: csharp/index.md (hunk 1 above)
Comment (pepone) [reply csharp/index.md]: move it to csharp/doc/index.md and it is used as the index of the generated doc

File: .gitignore (hunk 3 above)
Comment (pepone) [reply .gitignore]: yes fixed

**Pull request merged**